src/Silverback.Integration/Messaging/Diagnostics/ActivityConsumerBehavior.cs

```diff
@@ -42,20 +42,24 @@ public class ActivityConsumerBehavior : IConsumerBehavior
 
                 await next(context, serviceProvider).ConfigureAwait(false);
             }
-            finally
-            {
-                activity.Stop();
-            }
         }
 
         [SuppressMessage("", "CA1031", Justification = Justifications.ExceptionLogged)]
         private static void TryInitActivity(
             ConsumerPipelineContext context,
-            Activity activity,
+            Activity? activity,
             ISilverbackLogger? logger)
         {
+            if (activity == null || !activity.IsAllDataRequested)
+            {
+                return;
+            }
+
             try
             {
+                activity.AddTag("messaging.operation", "receive");
```

Comment:
Accoring to the Conventions we should add a lot more info. 
Also, some tag are specific for Kafka (eg. "messaging.kafka.consumer_group"). Should we add them as well?

Reply:
Uhm...I would say yes. But we have to implement _something_ to push different tags depending on the message broker implementation.
Something similar to the `AdditionalLogData` I use in the `SilverbackIntegrationLogger` to add the kafka key information to the log entries.

https://github.com/BEagle1984/silverback/blob/37600841196a94852656df24077e07d93f80efc4/src/Silverback.Integration.Kafka/Messaging/Behaviors/KafkaMessageKeyInitializerProducerBehavior.cs#L31

Or we could for example have a dedicated behavior that adds the additional tags after the `Activity` has been started. (I don't know if that's possible / if it's a good practice.)
This behavior should be implemented in `Silverback.Integration.Kafka`.

(I'm very open to a more elegant solution... 😞)

Reply:
I think adding tags later is not a problem. they just have to be there when the Activity ends. I would prefer that to the "bag" solution.

Reply:
OK, then a second behavior could definitely be added by Silverback.Integration.Kafka and triggered after this one, to add the extra tags to the current activity.